beam/www/beam/components/ControlButtons.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+	<div class="control-buttons">
+		<button class="btn" @click="onAmend" v-if="docstatus === 2">Amend</button>
+		<template v-else>
+			<button class="btn" @click="onSave" :disabled="!(docstatus === 0 && doctypeName === '')">Save</button>
+			<button class="btn" @click="onSubmit" :disabled="!(docstatus === 0 && doctypeName !== '')">Submit</button>
+			<button class="btn" @click="onCancel" :disabled="!(docstatus === 1)">Cancel</button>
+		</template>
+	</div>
+</template>
+
+<script setup lang="ts" generic="T extends ParentDoctype">
+import { ParentDoctype } from '@/types';
+import { defineProps, ref } from 'vue'
+const docstatus = ref(0)
+const doctypeName = ref("")
+
+const props = defineProps<{
+	onCreate: () => Promise<{ data: T }>
+	onSubmit: () => Promise<{ data: T; exception: string; response: Response }>
+	onCancel: () => Promise<{ data: T; exception: string; response: Response }>
+}>()
+
+const onSave = async () => {
+	try {
+		const { data } = await props.onCreate()
+		doctypeName.value = data.name
+		if (data) docstatus.value = data.docstatus
```

Comment:
This needs to be wrapped around the same if block:

```suggestion
		if (data) {
			doctypeName.value = data.name
			docstatus.value = data.docstatus
		}
```